app/controllers/annual_rent_updates_controller.rb

```diff
@@ -0,0 +1,22 @@
+class AnnualRentUpdatesController < ApplicationController
+  before_action :set_year, only: %i[show update destroy]
+  
+   # GET /annual-rent-updates
+  def index
+    response = use_case_factory.view_annual_rent_updates.execute
+    json_response(response)
+  end
+
+  # GET /annual-rent-updates/:year
+  def show
+    response = use_case_factory.view_annual_rent_updates.specific_year(@year)
+    return json_response(response) unless response[:error] == :no_record_found
```

Comment:
Yes. `json_response(response)` returns the error but it returns 200 for HTTP status . I used the guard clause to return 204 code for HTTP status 204 to show no content.

Reply:
but don't we want to return the error here if there is one? and you can still test the response in the spec if you were to allow the error to be returned by `json_response(response)`

Reply:
so you could have somethng like `expect(json_response['error']).to eq('no_record_found')` in your request spec. and you can still test the response as you are

Reply:
Yes. I suppose I can test the error in the 'request' test. But, I am already testing the error in the use case test, so I thought I can test the HTTP status code and return no response in the request test.

Reply:
Wouldn't returning `:no_record_found` be more useful than returning an empty response? Otherwise, what is the point of the usecase returning `:no_record_found` if the endpoint can never return it?

Reply:
I agree and I made the changes.